dotnet/src/webdriver/SeleniumManager.cs

```diff
@@ -41,24 +41,33 @@ static SeleniumManager()
             var currentDirectory = AppContext.BaseDirectory;
 
             string binary;
+            string file = "selenium-manager";
             if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
             {
-                binary = "selenium-manager/windows/selenium-manager.exe";
+                file = "selenium-manager.exe";
+                binary = $"selenium-manager/windows/{file}";
             }
             else if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
             {
-                binary = "selenium-manager/linux/selenium-manager";
+                binary = $"selenium-manager/linux/{file}";
             }
             else if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
             {
-                binary = "selenium-manager/macos/selenium-manager";
+                binary = $"selenium-manager/macos/{file}";
             }
             else
             {
                 throw new WebDriverException("Selenium Manager did not find supported operating system");
             }
 
-            binaryFullPath = Path.Combine(currentDirectory, binary);
+            if (Environment.GetEnvironmentVariable("SE_MANAGER_PATH") != null)
```

Comment:
I believe `SE_MANAGER_PATH` is full path to the sm binary. In this case let's restructure logic:
```csharp
if (env var exists)
{
    binaryFullPath = env var
}
else
{
    binaryFullPath = // use old logic
}
```

We need it to support the following requirement:
> using custom build on non-supported architecture

If we run "old logic", then we will get *Selenium Manager did not find supported operating system*.

Reply:
My thought was that it would be to the directory, not to the file (probably because that was the easiest way to do it in Ruby) 😂 
Let me think which is better. Good point on checking for OS support.